engine/src/main/java/com/google/android/fhir/sync/FhirDataStore.kt

```diff
@@ -100,6 +102,41 @@ internal class FhirDataStore(context: Context) {
     runBlocking { dataStore.edit { pref -> pref[lastSyncTimestampKey] = datetime.toString() } }
   }
 
+  private val mutexMap = mutableMapOf<String, Mutex>()
+
+  private fun getMutexForKey(key: String): Mutex {
+    return mutexMap.getOrPut(key) { Mutex() }
+  }
```

Comment:
Make this thread safe
```suggestion
  private val mutexMap = ConcurrentHashMap<String, Mutex>()

  private fun getOrCreateMutex(key: String): Mutex {
      // computeIfAbsent is thread-safe and only creates a Mutex if one doesn't exist.
      return mutexMap.computeIfAbsent(key) { Mutex() }
  }
```

Reply:
done.

Reply:
done.